examples/apache-beam/src/test/java/org/neo4j/importer/BeamExampleIT.java

```diff
@@ -0,0 +1,735 @@
+import org.neo4j.importer.v1.targets.EntityTarget;
+import org.neo4j.importer.v1.targets.NodeTarget;
+import org.neo4j.importer.v1.targets.PropertyMapping;
+import org.neo4j.importer.v1.targets.RelationshipTarget;
+import org.neo4j.importer.v1.targets.Target;
+import org.testcontainers.containers.Neo4jContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class BeamExampleIT {
+
+    @Rule
+    public final TestPipeline pipeline = TestPipeline.create();
+
+    @ClassRule
+    public static Neo4jContainer<?> NEO4J = new Neo4jContainer<>(DockerImageName.parse("neo4j:5-enterprise"))
+            .withEnv("NEO4J_ACCEPT_LICENSE_AGREEMENT", "yes")
+            .withAdminPassword("letmein!");
+
+    @Test
+    public void imports_dvd_rental_data_set() throws Exception {
+        try (InputStream stream = this.getClass().getResourceAsStream("/specs/dvd_rental.yaml")) {
+            assertThat(stream).isNotNull();
+
+            try (var reader = new InputStreamReader(stream)) {
+                var specification = ImportSpecificationDeserializer.deserialize(reader);
+                var sourceOutputs = new HashMap<Source, PCollection<GenericRecord>>(
+                        specification.getSources().size());
+                var targetOutputs = new HashMap<String, PCollection<WriteCounters>>();
+                var sortedTargets = sortTargets(specification.getTargets().getAll());
+                sortedTargets.forEach(target -> {
+                    var source = specification.findSourceByName(target.getSource());
+                    assertThat(source).isInstanceOf(ParquetSource.class);
+
+                    var sourceRecords = sourceOutputs.computeIfAbsent(source, (src) -> {
+                        var parquetSource = (ParquetSource) src;
+                        return pipeline.apply(
+                                "[source %s] Read records".formatted(parquetSource.getName()),
+                                ParquetIO.parseGenericRecords(
+                                                (SerializableFunction<GenericRecord, GenericRecord>) record -> record)
+                                        .withCoder(GenericRecordCoder.create())
+                                        .from(parquetSource.uri()));
+                    });
+
+                    var targetOutput = sourceRecords
+                            .apply(
+                                    "[target %s] Wait for implicit dependencies".formatted(target.getName()),
+                                    Wait.on(implicitDependenciesOf(target, targetOutputs)))
+                            .setCoder(sourceRecords.getCoder())
+                            .apply(
+                                    "[target %s] Spread records across batches".formatted(target.getName()),
+                                    SpreadAcrossBatches.range(
+                                            Runtime.getRuntime().availableProcessors()))
```

Comment:
I'm not sure about this at all, but could we just use one of the stock transforms for this purpose, such as:

```suggestion
                            .apply("[target %s] Assign keys to records".formatted(target.getName()),
                                    WithKeys.of((SerializableFunction<GenericRecord, Integer>) input -> ThreadLocalRandom.current().nextInt(Runtime.getRuntime().availableProcessors())))
                            .setCoder(
                                    KvCoder.of(VarIntCoder.of(), sourceRecords.getCoder()))
 ``

Reply:
nice, my search-fu missed it yesterday

Reply:
fixed in 98d2c86e6bb0a38091d718f672e7786aae7db97f